fenicsprecice/fenicsprecice.py

```diff
@@ -396,8 +400,8 @@ def initialize(self, coupling_subdomain, read_function_space=None, write_object=
         self._precice_vertex_ids = self._interface.set_mesh_vertices(self._interface.get_mesh_id(
             self._config.get_coupling_mesh_name()), self._owned_vertices.get_coordinates())
 
-        if self._coupling_type is CouplingMode.UNI_DIRECTIONAL_READ_COUPLING or \
-                self._coupling_type is CouplingMode.BI_DIRECTIONAL_COUPLING:
+        if (self._coupling_type is CouplingMode.UNI_DIRECTIONAL_READ_COUPLING or
+                self._coupling_type is CouplingMode.BI_DIRECTIONAL_COUPLING) and self._is_parallel:
```

Comment:
Why do you add the `self._is_parallel` here?

Reply:
Calling the function `get_communication_map()` only makes sense if we have a uni-directional read coupling or a bi-directional coupling. For a uni-directional write coupling we would never call the function `read_data()` and hence generating a communication map is not necessary. This `if` condition covers the cases where we need to generate a communication map.

Reply:
:+1: